app/main.py

```diff
@@ -1,3 +1,78 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
```

Comment:
If you're using python 3.10+, I'd suggest using `|` instead of `Union`:
```suggestion
    def __add__(self, other: int | float | Distance) -> Distance:
```
The code is much shorter than.
Change everywhere if python 3.10+